src/nstime.c

```diff
@@ -201,6 +201,12 @@ nstime_get(nstime_t *time) {
 	clock_gettime(CLOCK_MONOTONIC, &ts);
 	nstime_init2(time, ts.tv_sec, ts.tv_nsec);
 }
+#elif defined(JEMALLOC_HAVE_CLOCK_GETTIME_NSEC_NP)
+#  define NSTIME_MONOTONIC true
+static void
+nstime_get(nstime_t *time) {
+	nstime_init(time, clock_gettime_nsec_np(CLOCK_UPTIME_RAW));
+}
```

Comment:
While `clock_gettime_nsec_np` is preferred, the way we use `mach_absolute_time` should also be updated like [the second example here](https://developer.apple.com/library/archive/qa/qa1398/_index.html) for correctness on older mac os:
```
static mach_timebase_info_data_t    sTimebaseInfo;
if ( sTimebaseInfo.denom == 0 ) {
        (void) mach_timebase_info(&sTimebaseInfo);
}
nstime_init(time, mach_absolute_time() * sTimebaseInfo.numer / sTimebaseInfo.denom);
```

Reply:
Fixed